### DISCUSSION TITLE
Push timeout

### DISCUSSION THREAD

**Discussion by cmh716** [TwiN/gatus#821] [Ideas]
The new Push-based (external) checks are great additions. I wonder if there are any options like in Uptime Kuma, where we can send an alert when a push event notification has not been seen in a defined period of time? It's kind of like a watchdog timer.

**Comment by jinnatar** [comment]
Was just searching for exactly this. It would allow for a TTL style check where an endpoint device is expected to send POSTs on a cadence and a TTL value is reset in Gatus whenever a true/false is received. If the TTL is exceeded, the endpoint goes into a failed state.

Example scenario:
- A low powered device that can't reliably run a webserver could instead send POSTs every 60 seconds, say from a watchdog script.
- external_endpoint configured with a TTL/timeout param set to `120s`
- If the device onboard watchdog detects a failure condition and POSTs a `false`, the endpoint is displayed as failed. This could use a "condition string" in the hover card along the lines of `success = false`. The TTL would still get reset back  to 120s.
- If the device onboard watchdog fails itself and stops sending the POSTs, the TTL would run out to 0 and display the endpoint as failed. The "condition string" could say `TTL timeout (120s)`

**Comment by TwiN** [comment]
Have a look at #741 - that's definitely a feature we want.
